codecov/commands/base.py

```diff
@@ -29,13 +34,13 @@ def get_command(self, namespace):
 class BaseInteractor:
     requires_service = True
 
-    def __init__(self, current_owner: Owner, service: str):
+    def __init__(self, current_owner: Owner, service: str, current_user: User = None):
+        self.current_user = current_user or AnonymousUser()
         self.current_owner = current_owner
         self.service = service
 
         if not self.service and self.requires_service:
             raise MissingService()
 
-        self.current_user = AnonymousUser()
-        if self.current_owner:
-            self.current_user = self.current_owner.user
+        if self.current_owner and not self.current_user.is_authenticated:
+            self.current_user = self.current_owner
```

Comment:
This is mixing types.  I think this could remain as:
```python
if self.current_owner:
    self.current_user = self.current_owner.user
```

Reply:
oh sorry lost with the new changes